keylime/signing.py

```diff
@@ -49,19 +49,18 @@ def verify_signature(key, sig, file):
 
     # PGP
     if key_header == "-----BEGIN PGP PUBLIC KEY BLOCK-----":
-        gpg = gnupg.GPG()
-        logger.debug("Importing GPG key")
-        gpg_imported = gpg.import_keys(key.decode("utf-8"))
-        if gpg_imported.count == 1:  # pylint: disable=E1101
-            logger.debug("GPG key successfully imported")
-        else:
-            raise Exception("Unable to import GPG key")
-
-        # The Python PGP library won't let you read a signature from memory, hence this hack.
-        with tempfile.NamedTemporaryFile() as temp_sig:
-            temp_sig.write(sig)
-            temp_sig.flush()
-            verified = gpg.verify_data(temp_sig.name, file)
+        verified = False
+        with tempfile.TemporaryDirectory() as gpg_homedir:
+            ctx = gpg.Context(home_dir=gpg_homedir)
```

Comment:
Is there a reason we require the temp dir?
What happens if `home_dir=None` is set?

Reply:
If we don't set one, it will use the engine default, which would be e.g. `~/.gnupg`. Looking again, this seems to be what the current code is doing/using, so I guess we could remove this temp dir?

Reply:
No, the temp dir is then better in this case.